middleware/varnish/detectors-varnish.tf

```
@@ -0,0 +1,135 @@
+    B = data('varnish.cache_miss', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('B')
+    signal = (A/(A+B)).fill(0).scale(100).publish('signal')
+    detect(when(signal < ${var.varnish_cache_hit_rate_threshold_warning})).publish('WARN')
+    detect(when(signal < ${var.varnish_cache_hit_rate_threshold_critical}) and (signal > ${var.varnish_cache_hit_rate_threshold_warning})).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is too low > ${var.varnish_cache_hit_rate_threshold_critical}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.detectors_disabled)
+    notifications         = coalescelist(var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+  rule {
+    description           = "is too low > ${var.varnish_cache_hit_rate_threshold_warning}"
+    severity              = "Warning"
+    detect_label          = "WARN"
+    disabled              = coalesce(var.detectors_disabled)
+    notifications         = coalescelist(var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+}
+
+# memory used
+resource "signalfx_detector" "varnish_memory_status" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Varnish Memory Status"
+
+  program_text = <<-EOF
+    A = data('varnish.s0.g_space', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('A')
+    B = data('varnish.s0.g_bytes', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('B')
+    signal = ((B*100)/A).fill(0).scale(100).publish('signal')
```

I could be wrong but I don't think you want to multiply 2 times by 100.

```suggestion
    signal = (B/A).scale(100).fill(0).publish('signal')
```